File: 004Fibonacci-Swap-MinMax/C/fib.c

#include <stdio.h>

const int LastValue = 10000;


void fib(const int last_num){
    int num1=1, num2=0;
    int tmp;
    if(last_num < 0)
        return;

    printf("%d\n", num2);
    while(num1 <= last_num){
        printf("%d\n", num1);
        tmp = num1 + num2;
        num2 = num1;
        num1 = tmp;
    }
}

int main(void){

    fib(LastValue);
    return 0;
}

File: 004Fibonacci-Swap-MinMax/Go/fib.go

package main

import "fmt"

func main(){
    fib(10000);
}

func fib(last_value int){
    num1 := 1
    num2 := 0

    if num2 > last_value{
        return
    }
    fmt.Println(num2)
    for ;num1 < last_value; {
        fmt.Println(num1)
        tmp := num1 + num2
        num2 = num1
        num1 = tmp
    }
}


